File: DSA/Array/0001_Two_Sum/solution.py
# 0001. Two Sum
# Difficulty : Easy
# Tags       : array, hash-table
# Solved on  : 2026-05-05
# LeetCode   : https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i,num in enumerate(nums):
            complement = target - num
            if complement in seen:
                return [seen[complement],i]
            seen[num] = i
        
        

File: DSA/Array/0001_Two_Sum/solution.cpp
// 0001. Two Sum
// Difficulty : Easy
// Tags       : array, hash-table
// Solved on  : 2026-04-23
// LeetCode   : https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        map<int,int> mp;
        for(int i = 0 ;i<nums.size();i++)
        {
            int num = nums[i];
            int complement = target - num;
            if (mp.find(complement) != mp.end()){
                res.push_back(mp[complement]);
                res.push_back(i);
                return res;
            }
            mp[num] = i;
        }
        return res;
    }
};